numpy/core/arrayprint.py

```diff
@@ -521,7 +513,10 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
-    if a.size == 0:
+    if a.shape == () and a.dtype.names is None:
```

Comment:
I guess my suggestion might be as easy as adding `if style and a.shape == () ...`